### PR TITLE
[AIRFLOW-3541] Add Avro logical type conversion to bigquery hook

### PR DESCRIPTION
**Pull request by BrechtDeVlieger**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

Add option to allow logical type conversion when loading to BigQuery. 
https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.load.useAvroLogicalTypes

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

Very trivial change. Hard to test without integration with BigQuery.

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by BrechtDeVlieger** [comment]
Why did the tests fail?

File: airflow/contrib/hooks/bigquery_hook.py [hunk 1]
```diff
@@ -1064,7 +1064,7 @@ def run_load(self,
             'DATASTORE_BACKUP': ['projectionFields'],
             'NEWLINE_DELIMITED_JSON': ['autodetect', 'ignoreUnknownValues'],
             'PARQUET': ['autodetect', 'ignoreUnknownValues'],
-            'AVRO': [],
+            'AVRO': ['useAvroLogicalTypes'],
```
Comment (zhongjiajie) [inline airflow/contrib/hooks/bigquery_hook.py:1067]: @BrechtDeVlieger seem is k8s error, you can pull requests again to try it

**Comment by GrantSheehan** [comment]
Any reason this is languishing? Anything I can do to help get it merged?

**Comment by ashb** [comment]
I'm not familiar with Avro/BQ but does this need to be optional? Is there a situation where this changes  or breaks the existing behaviour?

**Comment by Fokko** [comment]
As a committer on Avro as well, I think I can chip in here. I would say that having BQ to pick up the logical types is a behavior that we would like to use. For example, the physical type of a TIMESTAMP is an int (seconds since epoch), but we would still like to have BQ interpret this a timestamp instead of a plain integer.

**Review by Fokko — APPROVED** [review]

**Comment by Fokko** [comment]
@ashb I've pushed this to 1.10.3 as well.

**Pull request merged**